### PR TITLE
Handle DOIs in Brackets; Throw Informative Error for Duplicate New Paper Issues

### PR DESCRIPTION
**Pull request by rdvelazquez**
### Description of the proposed additions or changes

- Correctly parses DOIs from issue citations in brackets; this was the reason that #107 wasn't correctly associated with the paper (see: https://github.com/greenelab/covid19-review/issues/228#issuecomment-615933570)
- Throws an informative error when duplicate issues exist (two issues for the same paper); There should be a build failure on this PR because  #106 and #165 address the same paper. I'll mark one as a duplicate after this PR builds; I just want to keep them both open for now so we can see how it will look in CI. _(Thee two issues didn't previously cause problems because #165 was missed due to the brackets)_ (see: https://github.com/greenelab/covid19-review/pull/230#issuecomment-615933976)

### Related issues

#172 

### Suggested reviewers (optional)

@agitter

### CODE REVIEW

**Comment by rando2** [comment]
Hi @rdvelazquez this looks totally reasonable to me, though I will defer to Tony! I like that you are aware of the possibility that we could end up with more than two issues for the same DOI, which seems like something that could happen! Are you imagining that we might need to add that into the error message down the road?

**Comment by rdvelazquez** [comment]
Thanks for looking at this @rando2!

> I like that you are aware of the possibility that we could end up with more than two issues for the same DOI, which seems like something that could happen! Are you imagining that we might need to add that into the error message down the road?

I tried to make the error message fairly generic so that it could handle that situation. If there's more than one issue per DOI it just throws an error and then lists all the DOIs that had more than one paper. I haven't tested an instance of three plus issues for a single paper but it should handle that fine. 

I'll wait for @agitter's review.

File: build/litsearch/getInternalData.py [hunk 1]
```diff
@@ -267,6 +282,12 @@ def mergePaperDataFrames(dataFramesList):
         dataFramesList_DOI.append(df[validDOI])
         dataFramesList_NoDOI.append(df[invalidDOI])
 
+    # Check if there are issues with duplicate DOIs
+    for df in dataFramesList_DOI:
+        duplicateDOIs = getDuplicates(list(df.index))
```
Comment (agitter) [inline build/litsearch/getInternalData.py:287]: The pandas `duplicated` function could be used to get duplicate rows. However, this is single-purpose code, so I don't think it matters whether we use that or a specialized function.

File: build/litsearch/getInternalData.py [hunk 2]
```diff
@@ -267,6 +282,12 @@ def mergePaperDataFrames(dataFramesList):
...
+        if len(duplicateDOIs) > 0:
+            raise ValueError('The following paper(s) has/have duplicate issues:', duplicateDOIs)
```
Comment (agitter) [inline build/litsearch/getInternalData.py:289]: I agree that we want the build to fail if that are duplicates.  That will be a stronger signal to maintainers to close the duplicate issues.

**Pull request merged**